### PR TITLE
[Merged by Bors] - Fix: LDAP bind credentials secret resolution

### LINKED ISSUE #366: LDAP Secret resolution not working correctly
### Affected version

current

### Current and expected behavior

LDAP secret should be resolved with k8sSearch but the name is taken from the SecretClass

### Possible solution

We'll implement framework functionality for it

### Additional context

_No response_

### Environment

_No response_

### Would you like to work on fixing this bug?

None

### PR DESCRIPTION
**Pull request by fhennig**
# Description

fixes #366 

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by fhennig** [comment]
Current state: 
- LDAP_USER and LDAP_PASSWORD env vars are now set in the command/arg of the Trino container. 
- Bind volume creation has been moved
- TLS volume creation has been moved
- TLS verification config has been adapted.

It _should_ work now. Next step is testing.

**Comment by fhennig** [comment]
I've started a test run: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/48/

Note that this PR currently has a reduced test set (I wanted the test to complete faster)

**Comment by fhennig** [comment]
Next steps:
- get the framework code merged
- update the operator-rs dependency to not point to branch anymore
- update changelog
- run the full test suite

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ All notable changes to this project will be documented in this file.
 
 - Hive catalog now properly handles hive clusters with replicas > 1 ([#348]).
 - Role group selectors are now applied to the generated StatefulSets ([#360]).
+- LDAP bind credentials Secret resolution with k8sSearch now works correctly ([#367]).
```
Comment (sbernauer) [inline CHANGELOG.md:28]:
```suggestion
- LDAP bind credentials resolution from SecretClasses now works correctly ([#367]).
```

File: rust/operator-binary/src/config.rs [hunk 2]
```diff
@@ -83,20 +80,18 @@ pub fn password_authenticator_properties(
...
-                    }
-                    TlsVerification::Server(_) => {
-                        config.insert(
-                            LDAP_SSL_TRUST_STORE_PATH.to_string(),
-                            Some(format!("{}/{}", LDAP_TRUST_CERT_PATH, "ca.crt")),
-                        );
-                    }
+            if ldap.use_tls() {
+                if !ldap.use_tls_verification() {
+                    // use TLS but don't verify => not supported
```
Comment (sbernauer) [inline rust/operator-binary/src/config.rs:85]:
```suggestion
                    // Use TLS but don't verify LDAP server ca => not supported
```

File: rust/operator-binary/src/controller.rs [hunk 3]
```diff
@@ -652,22 +648,17 @@ fn build_rolegroup_statefulset(
     // we need to mount ldap bind credentials from the secret as env vars
     if let Some(auth) = authentication_config {
         match auth {
-            TrinoAuthenticationConfig::MultiUser { .. } => {}
+            TrinoAuthenticationConfig::MultiUser { .. } => {
+                cb_prepare.add_volume_mount("users", USER_PASSWORD_DATA_DIR_NAME);
+                cb_trino.add_volume_mount("users", USER_PASSWORD_DATA_DIR_NAME);
+                pod_builder.add_volume(
+                    VolumeBuilder::new("users")
+                        .with_empty_dir(Some(""), None)
```
Comment (sbernauer) [inline rust/operator-binary/src/controller.rs:656]: I would prefer `.with_empty_dir(None, None)` (as we don't want to set any medium) for the whole file (5 matches)

**Review by sbernauer — CHANGES_REQUESTED** [review]

**Review by sbernauer — APPROVED** [review]
LGTM % with_empty_dir comment

File: rust/operator-binary/src/controller.rs (hunk 3 above)
Comment (fhennig) [reply rust/operator-binary/src/controller.rs]: done

**Comment by fhennig** [comment]
bors merge

**Pull request closed without merging**